packages/website-eslint/typings/eslint.d.ts

```diff
@@ -0,0 +1,7 @@
+// if we don't declare this, then TS in VSCode will use its "helpfully fetched"
+// types and report errors in the IDE
```

Comment:
VSCode has a helpful feature where it'll automatically fetch types for you for your JS node module imports.
So it means that in the IDE you'll open the file and VSCode will tell TS it can look in its cache for module types for the `@types/eslint` package. It finds it, uses it, then will show errors because our `RuleModule` doesn't match the `@types/eslint` `RuleModule`.

But this behaviour only happens in the IDE, not the CLI - because the CLI can't use VSCode's cache, ofc, so it just uses `any` for the import. So adding this type tells TS that it cannot use VSCode's cache ever - stubbing it out permanently.

I hate it fo sho. I wanted to use the types from `@typescript-eslint/utils/ts-eslint`, but this package is supposed to be as light-weight as possible.

In a nutshell this package is how we bundle up our project for use on the web. So it's a bit weird and hacky - mocks out some modules and such. I'll raise a PR against main to clean some stuff up.